src/subcommand/export.rs

```diff
@@ -0,0 +1,407 @@
+use {
+  super::*,
+  crate::index::entry::InscriptionEntry,
+  indicatif::{ProgressIterator, ProgressStyle},
+  log::log_enabled,
+};
+
+#[derive(Debug, Parser)]
+pub(crate) struct Export {
+  #[clap(
+    long,
+    default_value = "inscriptions",
+    help = "Write to directory <OUTPUT_DIR>"
+  )]
+  output_dir: PathBuf,
+}
+
+impl Export {
+  pub(crate) fn run(&self, options: Options) -> Result {
+    let index = Arc::new(Index::open(&options)?);
+    index.update()?;
+    self.run_with_index(index)?;
+    Ok(())
+  }
+
+  fn run_with_index(&self, index: Arc<Index>) -> Result {
+    self.prepare_directory_tree()?;
+    let written_numbers = self.get_written_numbers()?;
+    let ids = index.get_inscriptions(None)?;
+    let index = index.as_ref();
+    for id in Export::add_progress_bar(ids.into_values()) {
+      Export::retry(|| {
+        let entry = index
+          .get_inscription_entry(id)?
+          .ok_or_else(|| anyhow!("inscription entry not found: {id}"))?;
```

Comment:
The `entry` is only looked up to get the inscription number. Is there a better way to do that?